README.md

```diff
@@ -44,59 +50,53 @@ cd generative_data_prep
+### Example
+```shell
+python3 -m generative_data_prep pipeline --input_file_path=<PATH TO DATASET FILE> --output_path=<PATH TO OUTPUT DIRECTORY> --pretrained_tokenizer=gpt2 --max_seq_length=1024 --input_packing_config='single::drop' --shuffle=on_RAM
+```
 
-Each line in the input file can be formatted as one of the following:
-- Plain text
-- `{"prompt": "", "completion": ""}`
-- `[{"prompt": "text...", "completion": "text..."}, {"prompt": "text...", "completion": "text..."}, {"prompt": "text...", "completion": "text..."}, ...]`
+</br>
 
+## Input
 
-## End to end data preparation
-The `generative_data_prep/data_prep/pipeline.py` script takes a single jsonline or text file as input, shuffles it, splits it into multiple train/dev/test, tokenizes the text, packs it into fixed sequence lengths and then converts it to HDF5 file format.
+The input file format must be either `.txt` or [`.jsonl`](https://jsonlines.org/). Depending on whether you want to do [fine-tuning](#fine-tuning) or [pretraining](#pretraining), your data will need to be in a specific format.
 
-### Example
-```python
-python3 -m generative_data_prep pipeline --input_file_path=path_to_jsonl.jsonl --output_path=path_to_output_directory --pretrained_tokenizer=gpt2 --max_seq_length=1024 --shuffle=on_RAM
-```
+### `.jsonl` Format
 
-### Output
-The `output_path` will contain all the tokenized HDF5 split files, and a directory called `tokenizer`. The directory path that is passed in as the `output_path` flag is where the final dataset is saved that can be used as input data to upload and run training. The `tokenizer` directory will be transferred to any output checkpoints that are saved by Sambastudio so the tokenizer can be used for inference. If you include the `keep_split_jsonls` flag, then the `output_path` will additionally contain a `splits` directory that saves the jsonl versions of the HDF5 files, meaning that splits/train_1_of_X.jsonl is the jsonl text version of train_1_of_X.hdf5.
+The JSON Lines format can be used for fine-tuning, or pre-training/continual pre-training. Each line in the `.jsonl` format should be a json object with a `prompt`, and `completion` element. For example:
 
-The output HDF5 files each contain two datasets:
-- \"input_ids\": sequences of tokens ids
-- \"token_type_ids\": describe the type of each token. The default id assignments are:
-  - id=0 for tokens in the prompt
-  - id=1 for tokens in the completion
-  - id=2 for \<eos\> tokens that serve as padding tokens (will not be trained to predict)
-  - id=3 for \<eos\> tokens at the end of articles, that define the attention boundary when training with article attention
+```
+{"prompt": "What did the fox do?", "completion": "The quick brown fox jumped over the lazy dog."}
+{"prompt": "How much wood does a woodchuck chuck?", "completion": "A woodchuck chucks 1000 wood."}
+{"prompt": "Who sells seashells by the sea shore?", "completion": "She sells seashells by the sea shore."}
+```
```

Comment:
We also support lists of prompt completion pairs, this is important, maybe mention that this guarantees the prompt completion pairs will be next to each other and not shuffled. If the prompt completions are on different lines they will be shuffled

<img width="799" alt="image" src="https://github.com/sambanova/generative_data_prep/assets/137233191/796c6e14-a064-412c-95f3-08ea6f490d67">